src/main/plugin/iso19139.ca.HNAP/process/source-add.xsl

```diff
@@ -22,6 +22,105 @@
   ~ Rome - Italy. email: geonetwork@osgeo.org
   -->
 
-<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0">
-  <xsl:import href="../../iso19139/process/source-add.xsl"/>
+<!--
+Stylesheet used to update metadata adding a reference to a source record.
+-->
+<xsl:stylesheet xmlns:gmd="http://www.isotc211.org/2005/gmd"
+                xmlns:gco="http://www.isotc211.org/2005/gco"
+                xmlns:xlink="http://www.w3.org/1999/xlink"
+                xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+                xmlns:geonet="http://www.fao.org/geonetwork"
+                version="2.0"
+                exclude-result-prefixes="#all">
+
+  <xsl:param name="siteUrl"/>
+  <xsl:param name="sourceUuid"/>
+  <xsl:param name="sourceUrl" select="''"/>
+  <xsl:param name="sourceTitle" select="''"/>
+
+  <!-- Do a copy of every nodes and attributes -->
+  <xsl:template match="@*|node()">
+    <xsl:copy>
+      <xsl:apply-templates select="@*|node()"/>
+    </xsl:copy>
+  </xsl:template>
+
+  <xsl:template match="gmd:MD_Metadata | *[contains(@gco:isoType, 'MD_Metadata')]" priority="2">
+    <xsl:copy>
+      <xsl:copy-of select="@*"/>
+      <xsl:apply-templates select="
+        gmd:fileIdentifier | gmd:language | gmd:characterSet | gmd:parentIdentifier | gmd:hierarchyLevel |
+        gmd:hierarchyLevelName | gmd:contact | gmd:dateStamp | gmd:metadataStandardName | gmd:metadataStandardVersion |
+        gmd:dataSetURI | gmd:locale | gmd:spatialRepresentationInfo | gmd:referenceSystemInfo | gmd:metadataExtensionInfo |
+        gmd:identificationInfo | gmd:contentInfo | gmd:distributionInfo"/>
+      <xsl:choose>
+        <xsl:when test="gmd:dataQualityInfo">
+          <xsl:apply-templates select="gmd:dataQualityInfo"/>
+        </xsl:when>
+        <xsl:otherwise>
+          <gmd:dataQualityInfo>
+            <gmd:DQ_DataQuality>
+              <gmd:scope>
+                <gmd:DQ_Scope>
+                  <gmd:level>
+                    <gmd:MD_ScopeCode codeListValue="dataset"
+                                      codeList="http://nap.geogratis.gc.ca/metadata/register/napMetadataRegister.xml#IC_108"/>
```

Comment:
The dataset value in HNAP corresponds to the code `RI_622`. Also updated the codeList URI:

```suggestion
                    <gmd:MD_ScopeCode codeListValue="RI_622"                     
                    codeList="https://schemas.metadata.geo.ca/register/napMetadataRegister.xml#IC_108">dataset; jeuDonnées</gmd:MD_ScopeCode>
```